include/libsemigroups/pbr.hpp

```diff
@@ -314,56 +314,166 @@ namespace libsemigroups {
 
-    //! Insertion operator
+    //! \brief Insertion operator
     //!
-    //! This member function allows PBR objects to be inserted into an \ostream.
+    //! This member function allows PBR objects to be inserted into a
+    //! std::ostream.
     friend std::ostream& operator<<(std::ostream&, PBR const&);
 
    private:
     std::vector<std::vector<uint32_t>> _vector;
   };
 
-  //! Validate a PBR.
+  //! \brief Namespace for PBR helper functions.
   //!
-  //! This function throws a LibsemigroupsException if
-  //! the argument \p x is not valid.
+  //! This namespace contains helper functions for the PBR class.
+  namespace pbr {
+    //! \brief Returns the identity PBR with specified degree.
+    //!
+    //! This function returns a new PBR with degree equal to \p n where every
+    //! value is adjacent to its negative. Equivalently, \f$i\f$ is adjacent
+    //! \f$i + n\f$ and vice versa for every \f$i\f$ less than the degree
+    //! \f$n\f$.
+    //!
+    //! \param n the degree.
+    //!
+    //! \returns
+    //! A PBR.
+    //!
+    //! \exceptions
+    //! \no_libsemigroups_except
+    PBR one(size_t n);
+
+    // TODO(later) analogue of bipartition::underlying_partition?
+
+    //! \brief Validate a PBR.
+    //!
+    //! This function throws a LibsemigroupsException if the argument \p x is
+    //! not valid.
+    //!
+    //! \param x the PBR to validate.
+    //!
+    //! \throws LibsemigroupsException if any of the following hold:
+    //! * \p x does not describe a binary relation on an even number of points;
+    //! * \p x has a point related to a point that is greater than degree()
+    //! * a list of points related to a point is not sorted.
+    //!
+    //! \complexity
+    //! Linear in the PBR::degree x.
+    void validate(PBR const& x);
+
+  }  // namespace pbr
+
+  //! \brief Construct and validate a \ref PBR.
+  //!
+  //! Construct and validate a \ref PBR.
+  //!
+  //! \tparam T the types of the arguments.
   //!
-  //! \param x the PBR to validate.
+  //! \param args the arguments to forward to the \ref PBR constructor.
   //!
   //! \returns
-  //! (None)
+  //! A PBR constructed from \p args and validated.
   //!
-  //! \throws LibsemigroupsException if any of the following hold:
-  //! * \p x does not describe a binary relation on an even number of points;
-  //! * \p x has a point related to a point that is greater than degree()
-  //! * a list of points related to a point is not sorted.
+  //! \throws LibsemigroupsException if libsemigroups::validate(PBR const&)
+  //! throws when called with the constructed PBR.
   //!
-  //! \complexity
-  //! Linear in the PBR::degree of \p x.
-  void validate(PBR const& x);
+  //! \sa
+  //! \ref libsemigroups::PBR().
+  //!
+  //! \warning
+  //! No checks are performed on the validity of \p args prior to the
+  //! construction of the PBR object.
+  template <typename... T>
+  inline PBR to_pbr_no_checks(T... args) {
```

Comment:
Should be just to_pbr

Reply:
Is that the case for the other `to_pbr` functions, or just the template ones? I was getting linking errors, so I added the inlines but I forgot templates behave differently.